cmd/gapir/cc/main.cpp

```diff
@@ -669,8 +677,9 @@ void android_main(struct android_app* app) {
   std::string socket_file_path = internal_data_path + "/" + std::string(pipe);
   std::string uri = std::string("unix://") + socket_file_path;
   std::unique_ptr<Server> server = nullptr;
-  MemoryManager memoryManager(memorySizes);
-  auto cache = InMemoryResourceCache::create(memoryManager.getTopAddress());
+  std::shared_ptr<MemoryAllocator> allocator = createAllocator();
+  MemoryManager memoryManager(allocator);
+  auto cache = InMemoryResourceCache::create(allocator, RAM_CACHE_SIZE);
```

Comment:
Don't we want this to be the actual size of the allocator, rather than some subset of? i.e. if the allocator is 128MB, does it make sense to have a 256MB created cache?

Reply:
The cache will keep itself below this figure, but as the allocator fills up and starts failing to allocate regions, the cache will naturally level out regardless of what this figure is.

Reply:
Fair. Does it make sense to make this figure as large as possible (i.e. we want to make the cache take up as much space as we can)

Reply:
Yeah, that's why it's set to the size of the allocator. I guess we could set it to ~0, but it will make no practical difference.

Reply:
If you wanted to put two caches into the same allocator and stop one from starving out the other, you could use a number smaller than the size of the allocator to set a ceiling on a specific cache's usage. Otherwise this parameter isn't of much use in the new design TBH.